src/shared_utilities/drivers.jl

```diff
@@ -713,6 +723,69 @@ function relative_humidity(
     return es / ea
 end
 
+"""
+    specific_humidity_from_dewpoint(dewpoint_temperature, temperature, air_pressure, earth_param_set)
+
+Estimates the specific humidity given the dewpoint temperature, temperature of the air
+in Kelvin, and air pressure in Pa,  along with the ClimaLand earth_param_set. This is useful 
+for creating the PrescribedAtmosphere - which needs specific humidity - from ERA5 reanalysis data.
+
+We first compute the relative humidity using the Magnus formula, then the saturated vapor pressure, and then
+we compute q from vapor pressure and saturated vapor pressure.
+
+For more information on the Magnus formula, see e.g. 
+Lawrence, Mark G. (1 February 2005). 
+"The Relationship between Relative Humidity and the Dewpoint Temperature in Moist Air: 
+A Simple Conversion and Applications". 
+Bulletin of the American Meteorological Society. 86 (2): 225–234.
+"""
+function specific_humidity_from_dewpoint(
+    T_dew_air::data_FT,
+    T_air::data_FT,
+    P_air::data_FT,
+    earth_param_set,
+) where {data_FT <: Real}
+    thermo_params = LP.thermodynamic_parameters(earth_param_set)
+    _T_freeze = LP.T_freeze(earth_param_set)
+    sim_FT = typeof(_T_freeze)
+    # Obtain the relative humidity. This function requires temperatures in Celcius
+    rh::sim_FT = rh_from_dewpoint(
+        sim_FT(T_dew_air) - _T_freeze,
+        sim_FT(T_air) - _T_freeze,
+    )
+
+    # In the future, we will use Thermodynamics functions for all of the below. Note that the type of T_air must match the type of the parameters in thermo_params.
+    esat::sim_FT = Thermodynamics.saturation_vapor_pressure(
+        thermo_params,
+        sim_FT(T_air),
+        Thermodynamics.Liquid(),
+    )
+
+    e = rh * esat
+    q = data_FT(0.622 * e / (P_air - 0.378 * e))
+    return q
+end
+
+"""
+    rh_from_dewpoint(Td_C, T_C)
+
+Returns the relative humidity given the dewpoint temperature in Celsius and the 
+air temperature in Celcius, using the Magnus formula.
```

Comment:
Celsius =)

Reply:
I know we're in the land of Fahrenheit but...

Reply:
"Celcius" is a common typo for several reasons:

Phonetic Similarity: The "s" sound in "Celsius" can lead people to mistakenly insert an "i" after the "l" when spelling it, making it look like "Celcius."

Familiarity with Other Words: Words like "delicious" or "vicious" have a similar ending, so people may unconsciously apply that pattern when spelling "Celsius."

Lack of Awareness: Since "Celsius" is derived from the name of a person (Anders Celsius), the correct spelling might not be as well-known, leading to more frequent errors.

Overall, it's an understandable mistake, but the correct form is "Celsius."

Reply:
> I know we're in the land of Fahrenheit but...

lol, Im slightly embarassed!!

Reply:
yes! I asked ChatGPT if this was a common typo